src/Workspaces/Remote/ServiceHub/UserOperationBooster.cs

```diff
@@ -33,8 +34,16 @@ public static UserOperationBooster Boost()
 
                 if (s_count == 1)
                 {
-                    // boost to normal priority
-                    Process.GetCurrentProcess().PriorityClass = ProcessPriorityClass.Normal;
+                    try
+                    {
+                        // boost to normal priority
+                        Process.GetCurrentProcess().PriorityClass = ProcessPriorityClass.Normal;
+                    }
+                    catch (Exception e) when (e is PlatformNotSupportedException || e is Win32Exception)
+                    {
+                        // The runtime does not support changing process priority, so just return a NOP booster.
```

Comment:
❗ We should latch in this state to avoid throwing more than one of these exceptions (or at most a small number if concurrently used at the start)

Reply:
Is thread safety a concern? Or is just adding a `bool priorityModificationSupported` enough?

Reply:
we also set this value in Dispose.  is there a reason we don't check for that?

Reply:
ah, i see, you return a nop.

Reply:
> Is thread safety a concern? 

1. the boosters are not shared.  so no threading concern there.
2. a latch would only ever go from "supported" to "not supported".  so the worst that happens in a few requests throw, but eventually you get into the not-supported state, and stop throwing.  i don't think we'd need a latch that guaranteed only one throw.

Reply:
> a latch would only ever go from "supported" to "not supported". so the worst that happens in a few requests throw, but eventually you get into the not-supported state, and stop throwing. i don't think we'd need a latch that guaranteed only one throw.

Exactly. This is what I meant by "...at most a small number..."